server/chat-plugins/help.ts

```diff
@@ -0,0 +1,469 @@
+import {LogViewer} from './chatlog';
+import {ROOMFAQ_FILE} from './room-faqs';
+const PATH = 'config/chat-plugins/help.json';
+const MINIMUM_LENGTH = 7;
+const helpRoom = Rooms.get('help');
+const REGEX_WHITELIST = ['miapi', 'annika', 'kris'];
+
+export let helpData: PluginData;
+
+try {
+	helpData = JSON.parse(FS(PATH).readIfExistsSync() || "{}");
+} catch (e) {
+	helpData = {
+		stats: {},
+		pairs: {},
+		disabled: false,
+		queue: [],
+	};
+}
+/**
+ * Terms commonly used in helping that should be ignored
+ * within question parsing (the Help#find function).
+ * */
+const COMMON_TERMS = [
+	'<<(.+)>>', '(do|use|type)( |)(``|)(/|!|//)(.+)(``|)', 'go to', '/rfaq (.+)', 'you can', Chat.linkRegex, 'click',
+	'need to',
+].map(item => new RegExp(item, "i"));
+
+/**
+ * A message filtered by the Help filter.
+ */
+interface LoggedMessage {
+	/** Message that's matched by the Help filter. */
+	message: string;
+	/** The FAQ that it's matched to. */
+	faqName: string;
+	/** The regex that it's matched to. */
+	regex: string;
+}
+/** Object of stats for that day. */
+interface DayStats {
+	matches?: LoggedMessage[];
+	total?: number;
+}
+interface PluginData {
+	/** Stats - filter match and faq that was matched - done day by day. */
+	stats?: {[k: string]: DayStats};
+	/** Word pairs that have been marked as a match for a specific FAQ. */
+	pairs: {[k: string]: string[]};
+	/** Whether or not the filter is disabled. */
+	disabled?: boolean;
+	/** Queue of suggested regex. */
+	queue?: string[];
+}
+
+export class HelpAnswerer {
+	roomFaqs: AnyObject;
+	disabled?: boolean;
+	queue: string[];
+	data: PluginData;
+	constructor(data: PluginData) {
+		this.data = data;
+		this.roomFaqs = this.loadFaqs();
+		this.queue = data.queue ? data.queue : [];
+
+		FS(ROOMFAQ_FILE).onModify(() => {
+			// refresh on modifications to keep it up to date
+			this.roomFaqs = this.loadFaqs();
+		});
+	}
+	find(question: string, user?: User) {
+		const faqs: string[] = Object.keys((this.roomFaqs || '{}'))
+			.filter(item => item.length >= 6 && !this.roomFaqs[item].startsWith('>'));
+		for (const term of COMMON_TERMS) {
+			if (term.test(question)) return null;
+		}
+		for (const faq of faqs) {
+			const match = this.match(question, faq);
+			if (match) {
+				if (user) {
+					const timestamp = Chat.toTimestamp(new Date()).split(' ')[1];
+					const log = `${timestamp} |c| ${user.name}|${question}`;
+					this.log(log, faq, match.regex);
+				}
+				return this.roomFaqs[match.faq];
+			}
+		}
+		return null;
+	}
+	visualize(question: string, hideButton?: boolean, user?: User) {
+		const response = this.find(question, user);
+		if (response) {
+			let buf = '';
+			buf += Utils.html`<strong>You said:</strong> ${question}<br />`;
+			buf += `<strong>Our automated reply:</strong> ${Chat.formatText(response)}`;
+			question = Utils.escapeHTML(question);
+			if (!hideButton) {
+				buf += `<hr /><button class="button" name="send" value="A: ${question}">`;
```

Comment:
```suggestion
			if (!hideButton) {
				buf += Utils.html`<hr /><button class="button" name="send" value="A: ${question}">`;
```
You should be able to use `Utils.html` here.